### ISSUE TITLE
Alexa reporting "Sorry, I'm having a problem accessing your Simon Says EU skill right now" - No Chime TTS

### ISSUE THREAD

**Issue by thisisbenwoo** [nimroddolev/chime_tts#122]
### System Health details

`## System Information

version | core-2024.4.3
-- | --
installation_type | Home Assistant OS
dev | false
hassio | true
docker | true
user | root
virtualenv | false
python_version | 3.12.2
os_name | Linux
os_version | 6.1.73-haos-raspi
arch | aarch64
timezone | Europe/Paris
config_dir | /config

<details><summary>Home Assistant Community Store</summary>

GitHub API | ok
-- | --
GitHub Content | ok
GitHub Web | ok
GitHub API Calls Remaining | 5000
Installed Version | 1.34.0
Stage | running
Available Repositories | 1467
Downloaded Repositories | 36
HACS Data | ok

</details>

<details><summary>AccuWeather</summary>

can_reach_server | ok
-- | --
remaining_requests | 27

</details>

<details><summary>Home Assistant Cloud</summary>

logged_in | true
-- | --
subscription_expiration | October 7, 2024 at 2:00 AM
relayer_connected | true
relayer_region | eu-central-1
remote_enabled | true
remote_connected | true
alexa_enabled | true
google_enabled | false
remote_server | eu-central-1-15.ui.nabu.casa
certificate_status | ready
instance_id | 1c44e163da574d8bba1a2ee4cfc33d25
can_reach_cert_server | ok
can_reach_cloud_auth | ok
can_reach_cloud | ok

</details>

<details><summary>Home Assistant Supervisor</summary>

host_os | Home Assistant OS 12.2
-- | --
update_channel | stable
supervisor_version | supervisor-2024.04.0
agent_version | 1.6.0
docker_version | 25.0.5
disk_total | 58.0 GB
disk_used | 33.3 GB
healthy | true
supported | true
board | rpi4-64
supervisor_api | ok
version_api | ok
installed_addons | Duck DNS (1.16.0), Home Assistant Google Drive Backup (0.112.1), Advanced SSH & Web Terminal (17.2.0), CEC Scanner (3.0), Samba share (12.3.1), Z-Wave JS (0.5.0), AppDaemon (0.16.4), Mosquitto broker (6.4.0), MQTT Explorer (browser-1.0.1)

</details>

<details><summary>Dashboards</summary>

dashboards | 8
-- | --
resources | 18
views | 6
mode | storage

</details>

<details><summary>Recorder</summary>

oldest_recorder_run | April 11, 2024 at 7:36 AM
-- | --
current_recorder_run | April 21, 2024 at 2:53 PM
estimated_db_size | 854.88 MiB
database_engine | sqlite
database_version | 3.44.2

</details>

<details><summary>Spotify</summary>

api_endpoint_reachable | ok
-- | --

</details>`

### Checklist

- [X] I have enabled debug logging for my installation.
- [X] I have filled out the issue template to the best of my ability.
- [X] This issue only contains 1 issue (if you have multiple issues, open one issue for each issue).
- [X] This issue is not a duplicate issue of currently [previous issues](https://github.com/nimroddolev/chime_tts/issues?q=is%3Aissue+label%3A%22Bug%22+)..

### Describe the issue

Chime TTS version: v1.1.0-beta4 (v1.0.0 just didn't work, so I'm trying the beta)
Alexa Media Player version: v4.9.2
(See notes at the bottom of this section for further info)

Objective: I'm testing Chime TTS to work with my Alexa Media Player and Echo Dot 3.
Why am I testing Chime TTS: 
1. I like that I can choose multiple TTS platforms and can announce in multiple languages as a result.
2. I have read that Chime TTS can get rid of the delay I am currently experiencing with the standard Alexa Media Player and especially the notify function.

My YAML looks like this (sorry not sure why my YAML is not formatting properly):
`service: chime_tts.say
target:
  entity_id: media_player.lr
data:
  message: Hello
  tts_platform: cloud
  cache: true
  audio_conversion: Alexa
`
After clicking Call Service, there is a green check mark indicating that the service ran successfully, but my Echo Dot 3 just says:

_Sorry, I'm having a problem accessing your Simon Says EU skill right now_

**Note1**: I'm in France and my Alexa is connected to a French Alexa account 
- I have not tried it with connecting it to a US account for a number of reasons, and prefer not to.
- With that said, I would think that others in Europe may have the same problem.

**Note2**: When I go to the list of Skills available in the Alexa app, there is **no** _Simon Says_ skill
**Note3**: Calling the notify_alexa_media_lr using the following _**does**_ work (this is copied from a script announcing my dishwasher's done:

`alias: "Announcement: Dishwasher is Done"
sequence:
  \- service: media_player.play_media
    data:
      media_content_id: buzzers_pistols_01
      media_content_type: sound
    target:
      entity_id:
        \- media_player.partout
  \- service: notify.alexa_media_partout
    data:
      data:
        type: announce`

### Reproduction steps

1. Go to `Developer tools | Services`
2. Using UI or `YAML` above
3. Click `Call Service`

I have also cleared the cache several times to see if that is the problem.
Chime TTS creates the mp3 files correctly in the .../config/www/chimes_tts directory
The permissions seem correct also:
`
-rw-r--r--    1 root     root        9.7K Apr 21 15:12 el3ei6pc.mp3
`

### Debug logs

```text
2024-04-21 14:53:47.321 WARNING (SyncWorker_2) [homeassistant.loader] We found a custom integration chime_tts which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2024-04-21 15:11:04.337 DEBUG (MainThread) [custom_components.chime_tts] ----- Chime TTS Say Called. Version v1.1.0-beta4 -----
2024-04-21 15:11:04.338 DEBUG (MainThread) [custom_components.chime_tts.queue_manager] Adding service call to queue
2024-04-21 15:11:04.343 DEBUG (MainThread) [custom_components.chime_tts.helpers.helpers] ----- General Parameters -----
2024-04-21 15:11:04.344 DEBUG (MainThread) [custom_components.chime_tts.helpers.helpers] * entity_ids = ['media_player.lr']
2024-04-21 15:11:04.344 DEBUG (MainThread) [custom_components.chime_tts.helpers.helpers] * cache = True
2024-04-21 15:11:04.344 DEBUG (MainThread) [custom_components.chime_tts.helpers.helpers] * offset = 450.0
2024-04-21 15:11:04.344 DEBUG (MainThread) [custom_components.chime_tts.helpers.helpers] * final_delay = 0.0
2024-04-21 15:11:04.344 DEBUG (MainThread) [custom_components.chime_tts.helpers.helpers] * media_players_array = [{'entity_id': 'media_player.lr', 'platform': 'alexa_media', 'should_change_volume': False, 'initial_volume_level': -1.0, 'playback_volume_level': -1, 'group_members_supported': False, 'announce_supported': False, 'is_playing': False}]
2024-04-21 15:11:04.345 DEBUG (MainThread) [custom_components.chime_tts.helpers.helpers] * message = Hello
2024-04-21 15:11:04.345 DEBUG (MainThread) [custom_components.chime_tts.helpers.helpers] * tts_platform = cloud
2024-04-21 15:11:04.345 DEBUG (MainThread) [custom_components.chime_tts.helpers.helpers] * tts_speed = 100.0
2024-04-21 15:11:04.345 DEBUG (MainThread) [custom_components.chime_tts.helpers.helpers] * tts_pitch = 0
2024-04-21 15:11:04.345 DEBUG (MainThread) [custom_components.chime_tts.helpers.helpers] * announce = False
2024-04-21 15:11:04.345 DEBUG (MainThread) [custom_components.chime_tts.helpers.helpers] * fade_audio = False
2024-04-21 15:11:04.345 DEBUG (MainThread) [custom_components.chime_tts.helpers.helpers] * volume_level = -1
2024-04-21 15:11:04.346 DEBUG (MainThread) [custom_components.chime_tts.helpers.helpers] * join_players = False
2024-04-21 15:11:04.346 DEBUG (MainThread) [custom_components.chime_tts.helpers.helpers] * unjoin_players = False
2024-04-21 15:11:04.346 DEBUG (MainThread) [custom_components.chime_tts.helpers.helpers] * audio_conversion = -y -ac 2 -codec:a libmp3lame -b:a 48k -ar 24000 -write_xing 0
2024-04-21 15:11:04.347 DEBUG (MainThread) [custom_components.chime_tts] *** Checking Chime TTS audio cache ***
2024-04-21 15:11:04.358 DEBUG (MainThread) [custom_components.chime_tts] Cached audio found:
2024-04-21 15:11:04.358 DEBUG (MainThread) [custom_components.chime_tts] - public_path = "http://aaa.bbb.ccc.ddd:8123/local/chime_tts/el3ei6pc.mp3"
2024-04-21 15:11:04.358 DEBUG (MainThread) [custom_components.chime_tts] - audio_duration = 1.56
2024-04-21 15:11:04.358 DEBUG (MainThread) [custom_components.chime_tts] - local_path = None
2024-04-21 15:11:04.358 DEBUG (MainThread) [custom_components.chime_tts] - media_content_id = None
2024-04-21 15:11:04.358 DEBUG (MainThread) [custom_components.chime_tts] Apply audio conversion
2024-04-21 15:11:04.359 DEBUG (MainThread) [custom_components.chime_tts.helpers.helpers] Converting audio: "ffmpeg -i /config/www/chime_tts/el3ei6pc.mp3 -y -ac 2 -codec:a libmp3lame -b:a 48k -ar 24000 -write_xing 0 /config/www/chime_tts/el3ei6pc_converted.mp3"
2024-04-21 15:11:04.531 DEBUG (MainThread) [custom_components.chime_tts] *** Pre-Playback Actions ***
2024-04-21 15:11:04.531 DEBUG (MainThread) [custom_components.chime_tts] *** Chime TTS playback ***
2024-04-21 15:11:04.532 DEBUG (MainThread) [custom_components.chime_tts] 1 Alexa media player detected:
2024-04-21 15:11:04.532 DEBUG (MainThread) [custom_components.chime_tts] - media_player.lr
2024-04-21 15:11:04.532 DEBUG (MainThread) [custom_components.chime_tts] Calling notify.alexa_media with data:
2024-04-21 15:11:04.532 DEBUG (MainThread) [custom_components.chime_tts] - message: <audio src="http://aaa.bbb.ccc.ddd:8123/local/chime_tts/el3ei6pc.mp3"/>
2024-04-21 15:11:04.532 DEBUG (MainThread) [custom_components.chime_tts] - data: {'type': 'tts'}
2024-04-21 15:11:04.532 DEBUG (MainThread) [custom_components.chime_tts] - target: ['media_player.lr']
2024-04-21 15:11:04.534 DEBUG (MainThread) [custom_components.chime_tts] - Waiting 1.56s for audio playback to complete...
2024-04-21 15:11:06.098 DEBUG (MainThread) [custom_components.chime_tts.helpers.media_player] - Waiting until 1 media_player isn't playing...
2024-04-21 15:11:06.099 DEBUG (MainThread) [custom_components.chime_tts.helpers.media_player] ✔ media_player.lr
2024-04-21 15:11:06.099 DEBUG (MainThread) [custom_components.chime_tts] ----- Chime TTS Say Completed in 1.76 s -----
2024-04-21 15:12:55.516 DEBUG (MainThread) [custom_components.chime_tts] ----- Chime TTS Say Called. Version v1.1.0-beta4 -----
2024-04-21 15:12:55.516 DEBUG (MainThread) [custom_components.chime_tts.queue_manager] Adding service call to queue
2024-04-21 15:12:55.521 DEBUG (MainThread) [custom_components.chime_tts.helpers.helpers] ----- General Parameters -----
2024-04-21 15:12:55.521 DEBUG (MainThread) [custom_components.chime_tts.helpers.helpers] * entity_ids = ['media_player.lr']
2024-04-21 15:12:55.522 DEBUG (MainThread) [custom_components.chime_tts.helpers.helpers] * cache = True
2024-04-21 15:12:55.522 DEBUG (MainThread) [custom_components.chime_tts.helpers.helpers] * offset = 450.0
2024-04-21 15:12:55.522 DEBUG (MainThread) [custom_components.chime_tts.helpers.helpers] * final_delay = 0.0
2024-04-21 15:12:55.522 DEBUG (MainThread) [custom_components.chime_tts.helpers.helpers] * media_players_array = [{'entity_id': 'media_player.lr', 'platform': 'alexa_media', 'should_change_volume': False, 'initial_volume_level': -1.0, 'playback_volume_level': -1, 'group_members_supported': False, 'announce_supported': False, 'is_playing': False}]
2024-04-21 15:12:55.522 DEBUG (MainThread) [custom_components.chime_tts.helpers.helpers] * message = Hello
2024-04-21 15:12:55.522 DEBUG (MainThread) [custom_components.chime_tts.helpers.helpers] * tts_platform = cloud
2024-04-21 15:12:55.523 DEBUG (MainThread) [custom_components.chime_tts.helpers.helpers] * tts_speed = 100.0
2024-04-21 15:12:55.523 DEBUG (MainThread) [custom_components.chime_tts.helpers.helpers] * tts_pitch = 0
2024-04-21 15:12:55.523 DEBUG (MainThread) [custom_components.chime_tts.helpers.helpers] * announce = False
2024-04-21 15:12:55.523 DEBUG (MainThread) [custom_components.chime_tts.helpers.helpers] * fade_audio = False
2024-04-21 15:12:55.523 DEBUG (MainThread) [custom_components.chime_tts.helpers.helpers] * volume_level = -1
2024-04-21 15:12:55.523 DEBUG (MainThread) [custom_components.chime_tts.helpers.helpers] * join_players = False
2024-04-21 15:12:55.523 DEBUG (MainThread) [custom_components.chime_tts.helpers.helpers] * unjoin_players = False
2024-04-21 15:12:55.523 DEBUG (MainThread) [custom_components.chime_tts.helpers.helpers] * audio_conversion = -y -ac 2 -codec:a libmp3lame -b:a 48k -ar 24000 -write_xing 0
2024-04-21 15:12:55.524 DEBUG (MainThread) [custom_components.chime_tts] *** Checking Chime TTS audio cache ***
2024-04-21 15:12:55.535 DEBUG (MainThread) [custom_components.chime_tts] Cached audio found:
2024-04-21 15:12:55.536 DEBUG (MainThread) [custom_components.chime_tts] - public_path = "http://aaa.bbb.ccc.ddd:8123/local/chime_tts/el3ei6pc.mp3"
2024-04-21 15:12:55.536 DEBUG (MainThread) [custom_components.chime_tts] - audio_duration = 1.56
2024-04-21 15:12:55.536 DEBUG (MainThread) [custom_components.chime_tts] - local_path = None
2024-04-21 15:12:55.536 DEBUG (MainThread) [custom_components.chime_tts] - media_content_id = None
2024-04-21 15:12:55.536 DEBUG (MainThread) [custom_components.chime_tts] Apply audio conversion
2024-04-21 15:12:55.538 DEBUG (MainThread) [custom_components.chime_tts.helpers.helpers] Converting audio: "ffmpeg -i /config/www/chime_tts/el3ei6pc.mp3 -y -ac 2 -codec:a libmp3lame -b:a 48k -ar 24000 -write_xing 0 /config/www/chime_tts/el3ei6pc_converted.mp3"
2024-04-21 15:12:55.727 DEBUG (MainThread) [custom_components.chime_tts] *** Pre-Playback Actions ***
2024-04-21 15:12:55.727 DEBUG (MainThread) [custom_components.chime_tts] *** Chime TTS playback ***
2024-04-21 15:12:55.728 DEBUG (MainThread) [custom_components.chime_tts] 1 Alexa media player detected:
2024-04-21 15:12:55.728 DEBUG (MainThread) [custom_components.chime_tts] - media_player.lr
2024-04-21 15:12:55.728 DEBUG (MainThread) [custom_components.chime_tts] Calling notify.alexa_media with data:
2024-04-21 15:12:55.728 DEBUG (MainThread) [custom_components.chime_tts] - message: <audio src="http://aaa.bbb.ccc.ddd:8123/local/chime_tts/el3ei6pc.mp3"/>
2024-04-21 15:12:55.728 DEBUG (MainThread) [custom_components.chime_tts] - data: {'type': 'tts'}
2024-04-21 15:12:55.728 DEBUG (MainThread) [custom_components.chime_tts] - target: ['media_player.lr']
2024-04-21 15:12:55.730 DEBUG (MainThread) [custom_components.chime_tts] - Waiting 1.56s for audio playback to complete...
2024-04-21 15:12:57.293 DEBUG (MainThread) [custom_components.chime_tts.helpers.media_player] - Waiting until 1 media_player isn't playing...
2024-04-21 15:12:57.293 DEBUG (MainThread) [custom_components.chime_tts.helpers.media_player] ✔ media_player.lr
2024-04-21 15:12:57.293 DEBUG (MainThread) [custom_components.chime_tts] ----- Chime TTS Say Completed in 1.77 s -----
```

### Diagnostics dump

[config_entry-hacs-fb3857a92143fd8606b60c09990d2368.json](https://github.com/nimroddolev/chime_tts/files/15052057/config_entry-hacs-fb3857a92143fd8606b60c09990d2368.json)

**Comment by dbullendsl** [comment]
Have you checked your URL?  The best way I've found is to connect to it from outside your WiFi by pasting the URL in to your browser.  http://aaa.bbb.ccc.ddd:8123/local/chime_tts/el3ei6pc.mp3

The Simon says response is essentially generic meaning something went wrong, like inaccessible URL or audio parameters out of specs for the Amazon devices

**Comment by thisisbenwoo** [comment]
I used both my local IP eg. 192.168.1.55:8123/local/chime_tts/el3ei6pc.mp3 and it worked. I also used the nabu casa external address + /local/chime_tts/el3ei6pc.mp3 and it plays the mp3 correctly, but when I use it with the Echo, same result .. 

_Sorry, I'm having a problem accessing your Simon Says EU skill right now_

**Comment by dbullendsl** [comment]
To me it still seems you have a issue with the Alexa accessing the external url that Chime_TTS is providing. Before Chime_TTS came along I had to use the service notify.alexa_media and it was very particular that the URL was externally accessible to the alexa as they do not play files locally.

This is what I used to use:
```
service: notify.alexa_media
data:
  message: <audio src= "https://urlxxxx.org:443/local/mp3/audio_file.mp3"/>
  data:
    type: tts
  target:
    - media_player.kitchen_echo_dot
   ```
If this doesn't work by substituting the audio src bit with the generated URL into the service call above and the name of your media_player then something with your access is off.

**Comment by thisisbenwoo** [comment]
I don't disagree with what you're saying, but isn't this essentially the same as my original script below, and I'm trying to use Chime_TTS because I want to reduce/eliminate the delay I'm experiencing with Alexa. Or are you saying that the delay will go away because (a) the mp3 file is better and/or (b) perhaps having it cached, is what reduces the delay?

```
service: notify.alexa_media_xyz
data:
  data:
    type: announce
  message: Washing's done

```

Hopefully, not to complicate things, but I'm going to assume that the answer to (a) and (b) above is true, if I wanted to parse a dynamic set of words through Chime_TTS (i) would the generation of the mp3 file create a greater delay because of the fact that the mp3 file needs to be created and (ii) I'm not sure how I would pass the name of the mp3 file to the script.

I appreciate your help!

Ben

**Comment by dbullendsl** [comment]
I use this integration to do the same thing, but understand that using Alexas as your device there's always going to be a brief delay because they're not 'local' playback devices and depend on the cloud. In other words, they work by sending the announcement to the AWS platform and then back to your devices. 

This is my script to do what you're hoping to do I think.  It plays an alert that is the chime I found for LG washer/dryer and appends the voice notification. Eliminating the delay between two pieces, but not the inherent delay of sending to AWS and back.

```
service: chime_tts.say
metadata: {}
data:
  chime_path: /config/www/mp3/lg_signal.mp3
  message: The washer has finished its cycle.
  tts_platform: tts.piper
  volume_level: 0.4
  audio_conversion: Alexa
target:
  entity_id:
    - media_player.kitchen_echo_show
    - media_player.master_bedroom_echo_dot
    - media_player.office_echo_dot
    - media_player.utility_room_echo_flex
```
 I use the chime_tts integration for quite a few automated announcements from the washer/dryer, garage door opening, doorbell, weather briefing, etc.

**Comment by thisisbenwoo** [comment]
Just tried your script- where are you located? Europe or the US? My issue is that I'm in Europe.
That's why it says "_Sorry, I'm having a problem accessing your Simon Says **EU** skill right now_"

**Comment by dbullendsl** [comment]
Ok, well I'm in US so that may be it, I'm not familiar with differences between the US and French platforms.  Are you able to use directly the notify.alexa_meda service? If so you might want to try the chime_tts: say_url service to generate an audio file combining the alert tone and the announcement then using it with the notify.alexa_media. There would probably be a small penalty of time - but not much.

```
  - service: chime_tts.say_url
    data:
      chime_path: bells
      end_chime_path: ba_dum_tss
      message: This is a test
      tts_platform: google_translate
      audio_conversion: Alexa
    response_variable: chime_tts
  - service: notify.alexa_media
    data:
      message: <audio src= "{{chime_tts.url}}"/>
      target: media_player.office_echo_dot
      data:
        type: tts
 ```
This uses the say_url function and then pushes it out to alexas

**Comment by nimroddolev** [comment]
Hi @thisisbenwoo.

I noticed that the URL in your debug logs is not secured (starting with `http` and not `https`).:
```
http://aaa.bbb.ccc.ddd:8123/local/chime_tts/el3ei6pc.mp3
```

This might be the issue preventing your Echo from accessing the file despite you being able to access it via your browser.

Do you have an SSL certificate installed for your HA instance?

**Comment by thisisbenwoo** [comment]
Hmmm, interesting ... I used my nabucasa link and it worked, but with the local IP, there is no https. Let me try again ...

**Comment by thisisbenwoo** [comment]
Update ... I have a reverse proxy, so that I can access my Home Assistant instance over the Internet with https://ha.example.org. But I've gone through the docs several times over, and I am missing where I can tell Chime TTS to refer to https://ha.example.org, or did I miss something?

**Comment by nimroddolev** [comment]
Check that you have set your `external_url` to your Home Assistance instance in your `configuration.yaml` file, eg:

```
homeassistant:
  external_url: "https://YOUR_UNIQUE_ADDRESS.ui.nabu.casa/"
  internal_url: "http://192.168.X.X:8123/"
```

**Comment by thisisbenwoo** [comment]
It works! Thank you!

I love how Chime_TTS can work with multiple TTS engines and languages, but I also chose Chime_TTS because it was supposed to reduce the time from calling the service to the Echo saying the words. I still find that there is a delay of several (around 2.5 sec - I used a stopwatch) before i hear the words. I even tried using a negative offset, but it doesn't make any difference. BTW, do you want me to open a new ticket for this question?

Since I turned cache on, I tried the same call to the service several times. Same result. I am using Developer Tools | Service as I am trying the service.

My YAML is below:
```
service: chime_tts.say
data:
  offset: -600
  final_delay: 0
  message: Hello mate, how are you?
  tts_platform: google_translate
  volume_level: 1
  cache: true
  audio_conversion: Alexa
  language: en
  tts_speed: 125
  tld: com
target:
  entity_id: media_player.lr

```

**Comment by nimroddolev** [comment]
I'm glad to hear you got the integration working 👍 

Chime TTS cannot communicate faster than Home Assistant can to your Echo devices.

The delay reduction promise of Chime TTS is that it saves you from having to make 2 separate service calls to your Echo (one for the chime, and another for the TTS audio).
What Chime TTS does to reduce delays is create a single audio file with the chime and TTS combined, so that a single call is made instead of two.

**Comment by nimroddolev** [comment]
I am marking this issue as
